planter/planter.sh

```diff
@@ -16,50 +16,104 @@
 #
 # Use like: ./planter/planter.sh bazel build //cmd/...
 #
+# Environment variable options:
 # - $TAG can be overridden to choose a bazel version eg `TAG=0.8.0 planter.sh ...`
+# - Alternatively $IMAGE or $IMAGE_NAME can be overridden to set the exact image
+# - $NO_PULL will disable pulling the image before running if set
 # - $DOCKER_EXTRA can be set to supply extra args in the docker call
 # - $DRY_RUN will trigger echoing the docker call instead of running it
-# - planter will always `docker pull $IMAGE` before running it, see below
 #
 # To build kubernetes, first you checkout github.com/kubernetes/kubernetes
 # to $GOPATH/src/k8s.io/kubernetes and github.com/kubernetes/test-infra
 # to $GOPATH/src/k8s.io/test-infra.
 #
 # Then you can build with:
-# $ cd $GOPATH/src/k8s.io/kubernetes`
-# $ $GOPATH/src/k8s.io/test-infra/planter/planter.sh make bazel-build`
+# $ cd $GOPATH/src/k8s.io/kubernetes
+# $ $GOPATH/src/k8s.io/test-infra/planter/planter.sh make bazel-release
+# 
+# or build a specific binary like:
+# $ cd $GOPATH/src/k8s.io/kubernetes
+# $ $GOPATH/src/k8s.io/test-infra/planter/planter.sh bazel build //cmd/kubectl
 #
 
 set -o errexit
 set -o nounset
-IMAGE_NAME="gcr.io/k8s-testimages/planter"
+
+# these can be overridden but otherwise default to the current stable image
+# used to build kubernetes from the master branch
+IMAGE_NAME="${IMAGE_NAME:-gcr.io/k8s-testimages/planter}"
 TAG="${TAG:-0.14.0}"
-IMAGE="${IMAGE_NAME}:${TAG}"
+IMAGE=${IMAGE:-${IMAGE_NAME}:${TAG}}
+
 # We want to mount our bazel workspace and the bazel cache
 # - WORKSPACE is assumed to be in your current git repo, or alternatively $PWD
 REPO=$(git rev-parse --show-toplevel 2>/dev/null || true)
 REPO=${REPO:-${PWD}}
 
 # Ensure the bazel cache directory exists, as otherwise docker will create it,
-# owned by root.
+# possibly with the wrong owner.
 BAZEL_CACHE="${HOME}/.cache/bazel"
-[[ -d "${BAZEL_CACHE}" ]] || mkdir -p "${BAZEL_CACHE}"
-
-# Instead of mounting the full user ${HOME} which previously causes many issues like #5607 & #5736
-# We mount only the following folders:
-# - ${HOME}/.cache/bazel to share bazel cache across builds
-# - /tmp also needs to be a suitable tmpfs mounted with exec so that bazel
-# can use it when executing various things
-VOLUMES="-v ${REPO}:${REPO} -v ${BAZEL_CACHE}:${BAZEL_CACHE} --tmpfs /tmp:exec,mode=777"
-
-# We want to run as the host user so they own the build outputs etc.
-# Part of this is handled in planter/entrypoint.sh
-GID="$(id -g ${USER})"
-ENV="-e USER=${USER} -e GID=${GID} -e UID=${UID} -e HOME=${HOME}"
-# construct the final docker command, with SELinux disabled for this container
-CMD="docker pull ${IMAGE} && docker run --security-opt label:disable --rm ${VOLUMES} --user ${UID}:${GID} -w ${PWD} ${ENV} ${DOCKER_EXTRA:-} ${IMAGE} ${@}"
+if [ -d "${BAZEL_CACHE}" ] ; then
```

Comment:
I think this logic is backwards now - you want to create this directory only if it doesn't exist, but this will only create it if it does exist.

Reply:
ah yeah, will fix in a follow up 🤦‍♂️